client/executor/src/wasm_runtime.rs

```diff
@@ -341,7 +349,8 @@ fn create_versioned_wasm_runtime(
 		version,
 		heap_pages,
 		wasm_method,
-		instances: Default::default(),
+		instances: RwLock::new(SmallVec::with_capacity(max_instances)),
```

Comment:
I don't see the advantage of using `SmallVec` when we use `with_capacity` here. And @arkpar comment about cloning the instances isn't solved either.

Reply:
There won't be allocation for `max_instances <=8` since that's the stack-based size for `SmallVec`. But yeah, better just start without setting capacity. If `max_size` is greater than 8 it will allocate right away, even though we might never actually need all those instances.

Reply:
`with_capacity` allocates if `max_instances > 8`
https://docs.rs/smallvec/1.2.0/smallvec/struct.SmallVec.html#method.with_capacity